src/oisst/Increment/Increment.cc

```diff
@@ -23,130 +40,323 @@ namespace oisst {
+        fd(j) = fd_other(j);
+    }
+
     return *this;
   }
 
 // ----------------------------------------------------------------------------
 
-  Increment & Increment::operator -=(const Increment &) {
-    util::abor1_cpp("Increment::operator-= needs to be implemented.",
-                    __FILE__, __LINE__);
+  Increment & Increment::operator -=(const Increment &other) {
+    const int size = geom_->atlasFunctionSpace()->size();
+
+    for (int i = 0; i < vars_.size(); i++) {
+      auto fd       = make_view<float, 1>(atlasFieldSet_->field(0));
+      auto fd_other = make_view<float, 1>(other.atlasFieldSet()->field(0));
+      for (int j = 0; j < size; j++)
+        fd(j) -= fd_other(j);
+    }
+
     return *this;
   }
 
 // ----------------------------------------------------------------------------
 
-  Increment & Increment::operator +=(const Increment &) {
-    util::abor1_cpp("Increment::operator+= needs to be implemented.",
-                    __FILE__, __LINE__);
+  Increment & Increment::operator +=(const Increment &other) {
+    const int size = geom_->atlasFunctionSpace()->size();
+
+    for (int i = 0; i < vars_.size(); i++) {
+      auto fd       = make_view<float, 1>(atlasFieldSet_->field(0));
+      auto fd_other = make_view<float, 1>(other.atlasFieldSet()->field(0));
+      for (int j = 0; j < size; j++)
+        fd(j) += fd_other(j);
+    }
+
     return *this;
   }
 
 // ----------------------------------------------------------------------------
 
-  Increment & Increment::operator *=(const double &) {
-    util::abor1_cpp("Increment::operator*= needs to be implemented.",
-                    __FILE__, __LINE__);
+  Increment & Increment::operator *=(const double &zz) {
+    const int size = geom_->atlasFunctionSpace()->size();
+
+    for (int i = 0; i < vars_.size(); i++) {
+      auto fd       = make_view<float, 1>(atlasFieldSet_->field(0));
+      for (int j = 0; j < size; j++)
+        fd(j) *= zz;
+    }
+
     return *this;
   }
 
 // ----------------------------------------------------------------------------
 
   void Increment::accumul(const double & zz, const State & xx) {
-    util::abor1_cpp("Increment::accuml() needs to be implemented.",
-                     __FILE__, __LINE__);
+    auto fd = make_view<float, 1>(atlasFieldSet_->field(0));
+    auto fd_xx = make_view<float, 1>(xx.atlasFieldSet()->field(0));
+
+    const int size = geom_->atlasFunctionSpace()->size();
+
+    for (int i = 0; i < size; i++)
+      fd(i) += zz*fd_xx(i);
+
+    return;
   }
 
 // ----------------------------------------------------------------------------
 
-  void Increment::axpy(const double &, const Increment &, const bool check) {
-    util::abor1_cpp("Increment::axpy() needs to be implemented.",
-                    __FILE__, __LINE__);
+  void Increment::axpy(const double &zz, const Increment &dx, const bool check)
+  {
+    // Ligang: not sure what to do with "check"???
+//  ASSERT(!check || time_ == dx.validTime())
```

Comment:
```suggestion
  ASSERT(!check || time_ == dx.validTime())
```

This should be re-enabled. if `check == true` it should check to make sure the time windows are the same